pybamm/plotting/plot_voltage_components.py

```diff
@@ -34,6 +36,13 @@ def plot_voltage_components(
         Keyword arguments, passed to ax.fill_between
 
     """
+    # Check if the input is a Simulation and extract Solution
+    if isinstance(input_data, Simulation):
+        solution = input_data.solution
+    elif isinstance(input_data, Solution):
+        solution = input_data
+    else:
+        raise ValueError("Input must be a Solution or Simulation object")
```

Comment:
IMO specifying the type of the variable in the docs (docstring) should be enough.
```suggestion
```